test/e2e/cht-docker-utils.js

```diff
@@ -0,0 +1,135 @@
+const path = require('path');
+const fs = require('fs');
+const https = require('https');
+const { spawn } = require('child_process');
+const request = require('request-promise-native');
+
+const log = require('../../src/lib/log');
+
+const DEFAULT_PROJECT_NAME = 'cht_conf_e2e_tests';
+const dockerHelperDirectory = path.resolve(__dirname, '.cht-docker-helper');
+const dockerHelperScript = path.resolve(dockerHelperDirectory, './cht-docker-compose.sh');
+
+const downloadDockerHelperScript = () => new Promise((resolve, reject) => {
+  const file = fs.createWriteStream(dockerHelperScript, { mode: 0o755 });
+  https
+    .get('https://raw.githubusercontent.com/medic/cht-core/master/scripts/docker-helper-4.x/cht-docker-compose.sh', (response) => {
+      response.pipe(file);
+      file.on('finish', () => file.close(resolve));
+      file.on('error', () => file.close(reject));
+    })
+    .on('error', () => {
+      fs.unlinkSync(file.path);
+      file.close(() => reject('Failed to download CHT Docker Helper script "cht-docker-compose.sh"'));
+    });
+});
+
+const ensureScriptExists = async () => {
+  if (!fs.existsSync(dockerHelperDirectory)) {
+    await fs.promises.mkdir(dockerHelperDirectory);
+  }
+
+  if (!fs.existsSync(dockerHelperScript)) {
+    await downloadDockerHelperScript();
+  }
+};
+
+const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));
+
+const getProjectConfig = async (projectName) => {
+  try {
+    const configFile = await fs.promises.readFile(path.resolve(dockerHelperDirectory, `${projectName}.env`), 'utf8');
+    return Object.fromEntries(
+      configFile.toString()
+        .split('\n')
+        .map(line => line.split('='))
+        .filter(entry => entry.length === 2),
+    );
+  } catch (error) {
```

Comment:
Is there a particular workflow where you know we need this `catch`?  I am just not sure what the use is of recovering gracefully if we cannot find the config file....

Reply:
That's a good point, I figured we could try the default port in case the instance is somehow running but it makes more sense to throw early and let it crash in this unsupported scenario